imports/plugins/included/ui-search/lib/components/searchModal.js

```diff
@@ -20,6 +24,134 @@ class SearchModal extends Component {
     value: PropTypes.string
   }
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      allProducts: [],
+      productsBeforeSortAndFilter: [],
+      productsAfterSortAndFilter: [],
+      displaySortAndFilter: "",
+      showAllProducts: null,
+      lowPriceFilter: 0,
+      highPriceFilter: 0
+    };
+  }
+  componentWillMount() {
+    const allProducts = Collections.Products.find({ ancestors: [] }, { sort: { createdAt: -1 }, limit: 10 }).fetch();
+    let bool;
+    if (this.props.value && this.props.products.length > 0) {
```

Comment:
As `bool` value is set to true when `!this.props.value && allProducts.length > 0`, there's no need for so many `if  else statements`. That means lines 42 to 46 can be simplified as:
```js
let bool = false;
if (!this.props.value && allProducts.length > 0) {
  bool = true;
}